src/bin/clapper-app/clapper-app-uri-dialog.c

```diff
@@ -25,30 +25,30 @@
 #include "clapper-app-utils.h"
 
 static void
-_entry_text_changed_cb (GtkEntry *entry,
-    GParamSpec *pspec G_GNUC_UNUSED, AdwMessageDialog *dialog)
+_entry_text_changed_cb (AdwEntryRow *entry,
+    GParamSpec *pspec G_GNUC_UNUSED, AdwAlertDialog *dialog)
 {
-  GtkEntryBuffer *buffer = gtk_entry_get_buffer (entry);
-  guint text_length = gtk_entry_buffer_get_length (buffer);
+  const gchar *text = gtk_editable_get_text (GTK_EDITABLE (entry));
   gboolean enabled = FALSE;
 
-  if (text_length > 0) {
-    const gchar *text = gtk_entry_buffer_get_text (buffer);
+  if (strlen (text) > 0) {
     enabled = (text && gst_uri_is_valid (text));
```

Comment:
```suggestion
  if (text && *text != '\0') {
    enabled = gst_uri_is_valid (text);
```

Reply:
Since in updated code we no longer use `text_length` before obtaining `text` pointer, we need to check if text exists and is not an empty NULL-terminated string. Otherwise we are gonna segfault here after this callback will be called when text is removed.

Reply:
Done!